x-pack/metricbeat/module/meraki/device_health/switchports.go

```diff
@@ -0,0 +1,155 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package device_health
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	meraki "github.com/meraki/dashboard-api-go/v3/sdk"
+)
+
+type switchport struct {
+	port       *meraki.ResponseItemSwitchGetOrganizationSwitchPortsBySwitchPorts
+	portStatus *meraki.ResponseItemSwitchGetDeviceSwitchPortsStatuses
+}
+
+func getDeviceSwitchports(client *meraki.Client, organizationID string, devices map[Serial]*Device, period time.Duration) error {
+	switches, res, err := client.Switch.GetOrganizationSwitchPortsBySwitch(organizationID, &meraki.GetOrganizationSwitchPortsBySwitchQueryParams{})
```

Comment:
I am not aware about the Meraki service. So assuming all the API calls are correct. 
Looks like from the data.json of the metricset.

Reply:
hope so 🙈